controllers/articles-controllers.js

```diff
@@ -16,9 +17,21 @@ function getArticleById(request, response, next) {
 }
 
 function getArticles(request, response, next) {
-  fetchAllArticles().then((articles) => {
-    response.status(200).send(articles);
-  });
+  const { topic } = request.query;
+
+  if (topic) {
+    return fetchArticleByTopic(topic)
+      .then((articles) => {
+        response.status(200).send(articles);
+      })
+      .catch((error) => next(error));
+  } else {
+    fetchAllArticles()
```

Comment:
instead of having a separate model to filter by topic, could build up your query conditionally as in the notes for complex queries to saving repeating logic in these similar models